### PR TITLE
Fix for Upgrade DataService Version/Image

### PR DESCRIPTION
**Pull request by Madanagopal19**
aetos logs : 
http://aetos.pwx.purestorage.com/resultSet/testSetID/118763 - ImageUpgrade
http://aetos.pwx.purestorage.com/resultSet/testSetID/118127 - VersionUpgrade

### CODE REVIEW

File: drivers/pds/lib/utils.go [hunk 1]
```diff
@@ -630,7 +630,9 @@ func GetVersionsImage(dsVersion string, dsBuild string, dataServiceID string) (s
 	isVersionAvailable = false
 	isBuildAvailable = false
 	for i := 0; i < len(versions); i++ {
-		if (*versions[i].Enabled) && (*versions[i].Name == dsVersion) {
+		log.Debugf("version name %s and is enabled=%t", *versions[i].Name, *versions[i].Enabled)
+		if *versions[i].Name == dsVersion {
+			log.Debugf("DS Version %s is enabled in the control plane", dsVersion)
```
Comment (lsrinivas-pure) [inline drivers/pds/lib/utils.go:635]: here enable check is not happening . is it intentional?

File: tests/pds/dataservice_test.go [hunk 2]
```diff
@@ -911,6 +911,19 @@ func UpgradeDataService(dataservice, oldVersion, oldImage, dsVersion, dsBuild st
...
+		defer func() {
+			Step("Delete the workload generating deployments", func() {
+				if Contains(dataServiceDeploymentWorkloads, ds.Name) {
+					log.InfoD("Deleting Workload Generating pods %v ", dep.Name)
+					err = pdslib.DeleteK8sDeployments(dep.Name, namespace)
+				} else if Contains(dataServicePodWorkloads, ds.Name) {
+					log.InfoD("Deleting Workload Generating pods %v ", pod.Name)
+					err = pdslib.DeleteK8sPods(pod.Name, namespace)
+				}
+				log.FailOnError(err, "error deleting workload generating pods")
```
Comment (lsrinivas-pure) [inline tests/pds/dataservice_test.go:923]: add ds.Name in the error log

File: drivers/pds/lib/utils.go (hunk 1 above)
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go:635]: yes, we don't need that check as suggested by developers.

File: tests/pds/dataservice_test.go (hunk 2 above)
Comment (Madanagopal19) [reply tests/pds/dataservice_test.go]: Done

**Review by lsrinivas-pure — APPROVED** [review]
lgtm

**Pull request merged**